widget-store/src/utilities.ts

```diff
@@ -115,12 +113,34 @@ export class ShopUtilities {
   }
 
   @Transaction({ readOnly: true })
-  static async retrieveOrderDetails(ctxt: KnexTransactionContext, order_id: number): Promise<OrderWithProduct[]> {
-    const items = await ctxt.client<Order>('orders')
-      .join<Product>('products', 'orders.product_id', 'products.product_id')
-      .select('orders.*', 'products.product')
-      .where({order_id});
-    return items as OrderWithProduct[];
+  static async retrieveOrders(ctxt: KnexTransactionContext) {
+    return ctxt.client<Order>('orders').select("*");
+  }
+
+  @Transaction({ readOnly: true })
+  static async retrievePaidOrders(ctxt: KnexTransactionContext) {
+    return ctxt.client<Order>('orders').select("*").where({ order_status: OrderStatus.PAID });
+  }
+
+  @Transaction()
+  static async makeProgressOnPaidOrder(ctxt: KnexTransactionContext, order_id: number): Promise<void> {
+    const orders = await ctxt.client<Order>('orders').where({
+      order_id: order_id,
+      order_status: OrderStatus.PAID,
+    });
+    if (!orders.length) {
+      throw new Error(`No PAID order with ID ${order_id} found`);
+    }
+
+    const order = orders[0];
+    if (order.progress_remaining > 1) {
+      await ctxt.client<Order>('orders').where({ order_id: order_id }).update({ progress_remaining: order.progress_remaining - 1 });
+    } else {
+      await ctxt.client<Order>('orders').where({ order_id: order_id }).update({
+        order_status: OrderStatus.DISPATCHED,
+        progress_remaining: 0
+      });
+    }
```

Comment:
Can you check if `nightlyReport`, `getDailySales`, `sendStatusEmail`, and `SalesSummary` are doing anything?  If not, remove them.

Reply:
They do things as per the [readme](https://github.com/dbos-inc/dbos-demo-apps/tree/main/widget-store#nightly-sales-reports-optional), i.e. send out an email report IF you configure it. I did not touch those parts. Do we want to keep it, or remove?

Reply:
Let's leave it then